### PR TITLE
(fix #13822):Support for Custom Properties for various entities

### LINKED ISSUE #13822: support for most custom properties to right side panel
**Is your feature request related to a problem? Please describe.**

- [ ] It's good to have custom properties listed on Right side panel for Asset details page
- [ ] At most 5 custom properties need to be listed on right side panel (read more will redirect to respective tab)
- [ ] Also good to have right side panel as collapsible 

Screenshot 2023-11-02 at 11 44 30 AM

### PR DESCRIPTION
**Pull request by harsh-vador**
### Describe your changes:

Fixes #13822 

I worked on providing support for Custom Properties for the entities in Custom Property.

https://github.com/open-metadata/OpenMetadata/assets/58542468/36995cf1-6492-4b7c-9c01-660d30457d0f

#
### Type of change:

- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/cypress/e2e/Pages/Entity.spec.ts [hunk 1]
```diff
@@ -113,6 +113,7 @@ entities.forEach((entity) => {
 
       it(`Set ${properties} Custom Property `, () => {
         Object.values(CustomPropertyType).forEach((type) => {
+          // set and validate
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/cypress/e2e/Pages/Entity.spec.ts:116]: We can revert this comment

File: openmetadata-ui/src/main/resources/ui/cypress/e2e/Pages/Entity.spec.ts [hunk 2]
```diff
@@ -122,6 +123,7 @@ entities.forEach((entity) => {
 
       it(`Update ${properties} Custom Property`, () => {
         Object.values(CustomPropertyType).forEach((type) => {
+          // update and validate
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/cypress/e2e/Pages/Entity.spec.ts:126]: same

File: openmetadata-ui/src/main/resources/ui/src/components/Entity/EntityRightPanel/EntityRightPanel.tsx [hunk 3]
```diff
@@ -94,6 +104,30 @@ const EntityRightPanel: FC<EntityRightPanelProps> = ({
         {KnowledgeArticles && (
           <KnowledgeArticles entityId={entityId} entityType={entityType} />
         )}
+        {customProperties?.extension && (
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Entity/EntityRightPanel/EntityRightPanel.tsx:107]: Isn't `customProperties` itself contain extension details?

File: openmetadata-ui/src/main/resources/ui/src/pages/SearchIndexDetailsPage/SearchIndexDetailsPage.tsx [hunk 4]
```diff
@@ -346,80 +345,75 @@ function SearchIndexDetailsPage() {
...
-          ...searchIndexDetails,
-          extension: updatedData.extension,
-        }));
+        (await onSearchIndexUpdate(
+          {
+            ...searchIndexDetails,
+            extension: updatedData.extension,
+          },
+          'extension'
+        ));
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/SearchIndexDetailsPage/SearchIndexDetailsPage.tsx:354]: Don't we need to update the state post patch call here?

File: openmetadata-ui/src/main/resources/ui/src/components/Entity/EntityRightPanel/EntityRightPanel.tsx [hunk 5]
```diff
@@ -36,6 +42,8 @@ interface EntityRightPanelProps {
   domain?: EntityReference;
   onTagSelectionChange?: (selectedTags: EntityTags[]) => Promise<void>;
   onThreadLinkSelect?: (value: string, threadType?: ThreadType) => void;
+  viewAllPermission?: boolean;
+  customProperties?: Table['extension'];
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Entity/EntityRightPanel/EntityRightPanel.tsx:46]: Type of `customProperties` is Extention but we are using it as entityDetails can you double check here?

File: openmetadata-ui/src/main/resources/ui/cypress/e2e/Pages/Entity.spec.ts (hunk 1 above)
Comment (harsh-vador) [reply openmetadata-ui/src/main/resources/ui/cypress/e2e/Pages/Entity.spec.ts:116]: Done

File: openmetadata-ui/src/main/resources/ui/src/components/Entity/EntityRightPanel/EntityRightPanel.tsx (hunk 3 above)
Comment (harsh-vador) [reply openmetadata-ui/src/main/resources/ui/src/components/Entity/EntityRightPanel/EntityRightPanel.tsx:107]: Done

File: openmetadata-ui/src/main/resources/ui/cypress/e2e/Pages/Entity.spec.ts (hunk 2 above)
Comment (harsh-vador) [reply openmetadata-ui/src/main/resources/ui/cypress/e2e/Pages/Entity.spec.ts:126]: Done

File: openmetadata-ui/src/main/resources/ui/src/pages/SearchIndexDetailsPage/SearchIndexDetailsPage.tsx (hunk 4 above)
Comment (harsh-vador) [reply openmetadata-ui/src/main/resources/ui/src/pages/SearchIndexDetailsPage/SearchIndexDetailsPage.tsx:354]: This is being handled by the onSearchIndexUpdate function

File: openmetadata-ui/src/main/resources/ui/src/components/Entity/EntityRightPanel/EntityRightPanel.tsx (hunk 5 above)
Comment (harsh-vador) [reply openmetadata-ui/src/main/resources/ui/src/components/Entity/EntityRightPanel/EntityRightPanel.tsx]: Made necessary changes

**Pull request merged**